tests/test_standardizer.py

```diff
@@ -0,0 +1,357 @@
+import unittest
+import tempfile
+import warnings
+import os
+
+from astropy.io import fits as fitsio
+from astropy.time import Time
+import numpy as np
+
+from utils import DECamImdiffFactory
+from kbmod import PSF, Standardizer, StandardizerConfig
+from kbmod.standardizers import (
+    KBMODV1,
+    KBMODV1Config,
+    FitsStandardizer,
+    SingleExtensionFits,
+    MultiExtensionFits,
+)
+
+
+# Use a shared factory to skip having to untar the archive
+FitsFactory = DECamImdiffFactory()
+
+
+class MyStd(KBMODV1):
+    """Custom standardizer for testing Standardizer registration"""
+
+    name = "MyStd"
+    priority = 3
+    testing_kwargs = False
+
+    @classmethod
+    def yesStandardize(cls, tgt):
+        _, resources = super().resolveTarget(tgt)
+        return True, resources
+
+    @classmethod
+    def noStandardize(cls, tgt):
+        return False, {}
+
+    @classmethod
+    def resolveTarget(cls, tgt):
+        return cls.noStandardize(tgt)
+
+    def __init__(self, *args, required_flag, optional_flag=False, **kwargs):
+        super().__init__(*args, **kwargs)
+        if not self.testing_kwargs:
+            self.required_flag = False
+        else:
+            self.required_flag = required_flag
+        self.optional_flag = optional_flag
+
+    def translateHeader(self):
+        # invoke the parent functionality to standardize the default values
+        metadata = super().translateHeader()
+        metadata["required_flag"] = False
```

Comment:
Could the next few lines just be:
```
metadata["required_flag"] = self.required_flag
metadata["optional_flag"] = self.optional_flag
```

Reply:
That would make sense and would be much easier to read. Unfortunately I'm not sure I can do the `optional_flag` the same way. See below.